src/main/java/com/intuit/graphql/orchestrator/federation/EntityTypeMerger.java

```diff
@@ -22,7 +23,13 @@ public TypeDefinition mergeIntoBaseType(EntityMergingContext entityMergingContex
 
   private void merge(EntityMergingContext entityMergingContext) {
     List<FieldDefinition> baseTypeFields = getFieldDefinitions(entityMergingContext.getBaseType());
-    List<FieldDefinition> typeExtensionFields =  getFieldDefinitions(entityMergingContext.getTypeExtension());
+    List<FieldDefinition> typeExtensionFields = null;
+    if(entityMergingContext.getExtensionSystemDefinition().getType() != null) {
```

Comment:
I'm thinking of having a util function for this, e.g.

isTypeExtensionDefinition()